src/v/kafka/server/group_metadata.h

```diff
@@ -138,3 +311,130 @@ struct hash<kafka::offset_metadata_key> {
 };
 
 } // namespace std
+
+namespace reflection {
+
+/*
+ * new code + new version: will see extra bytes for client host/id
+ * new code + old version: will observe no extra bytes
+ * old code + old/new version: will not read new appended fields
+ *
+ * on the wire we write out a version that is compatible with old code and then
+ * append the additional client host/id information for each member. then when
+ * deserializing we process the appended data by updating the old versions from
+ * disk in the new in-memory structs which contain the extra member metadata.
+ * old code skips over the appended data.
+ */
+template<>
+struct adl<kafka::old::group_log_group_metadata> {
+    struct member_state_v0 {
+        kafka::member_id id;
+        std::chrono::milliseconds session_timeout;
+        std::chrono::milliseconds rebalance_timeout;
+        std::optional<kafka::group_instance_id> instance_id;
+        kafka::protocol_type protocol_type;
+        std::vector<kafka::member_protocol> protocols;
+        iobuf assignment;
+    };
```

Comment:
hmm, i think we need to stick to the `reflection` namespace. It can eventually be removed, maybe i can hide implementation in some other place. will try